integration/jnifuse/fs/src/main/java/alluxio/jnifuse/utils/NativeLibraryLoader.java

```diff
@@ -57,10 +62,113 @@ public static NativeLibraryLoader getInstance() {
   }
 
   /**
+   * Internal interface to write tryLoad in an elegant way
+   *
+   * @see Runnable interface cannot have checked exception
+   */
+  interface Load {
+    void load() throws IOException;
+  }
+
+  /**
+   * Try run the loading function.
+   *
+   * @param run the function to load library
+   * @return the error if UnsatisfiedLinkError is encountered, empty if it completes successfully.
+   * @throws IOException if a filesystem operation fails
+   */
+  private Optional<UnsatisfiedLinkError> tryLoad(Load run) throws IOException {
+    try {
+      run.load();
+      return Optional.empty();
+    } catch (final UnsatisfiedLinkError ule) {
+      return Optional.of(ule);
+    }
+  }
+
+  /**
+   * Load a version of libjnifuse.
+   * <p>
    * Firstly attempts to load the library from <i>java.library.path</i>,
    * if that fails then it falls back to extracting
    * the library from the classpath.
-   * {@link NativeLibraryLoader#loadLibraryFromJar(String)}
+   * {@link NativeLibraryLoader#loadLibraryFromJar(String, String, String)}
+   */
+  private Optional<UnsatisfiedLinkError> load(
+      final String sharedLibraryName, final String jniLibraryName,
+      final String sharedLibraryFileName, final String jniLibraryFileName,
+      final String tmpDir) throws IOException {
+
+    Optional<UnsatisfiedLinkError> err = tryLoad(() -> {
+      System.loadLibrary(sharedLibraryName);
+      LOG.info("Loaded {} by System.loadLibrary.", sharedLibraryName);
+    });
+
+    if (!err.isPresent()) {
```

Comment:
if err.isPresent return err?

Reply:
Fixed. Thanks!

Reply:
Empty means the load is successful now. No need to continue trying loading from other locations and just return an empty.